scripts/analytics/node_log_events.py

```diff
@@ -0,0 +1,92 @@
+import time
+import logging
+import argparse
+from modularodm import Q
+from datetime import datetime, timedelta
+from dateutil.parser import parse
+
+from website.app import init_app
+from website.project.model import NodeLog
+from website.settings import KEEN as keen_settings
+from keen.client import KeenClient
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+def get_events(date):
+    """ Get all node logs from a given date for a 24 hour period.
+    Defaults to starting yesterday and gathering through today (both in UTC).
+    """
+
+    # Make sure default is yesterday and not today
+    if date.day == datetime.utcnow().day:
+        date = date - timedelta(1)
+
+    logger.info('Gathering node logs between {} and {}'.format(
+        date.isoformat(), (date + timedelta(1)).isoformat())
+    )
+
+    node_log_query = Q('date', 'lt', date + timedelta(1)) & Q('date', 'gte', date)
+
+    node_logs = NodeLog.find(node_log_query)
+    node_log_events = []
+    for node_log in node_logs:
+        event = {
+            'keen': {'timestamp': date.isoformat()},
```

Comment:
I mentioned this in the summary comment: since this is an event, the timestamp should be the timestamp of the event itself rather than the given date object.